nvsvocprez/utils.py

```diff
@@ -250,4 +253,92 @@ def get_alt_profiles() -> Dict:
         return altprof_data_by_url
     except requests.RequestException as exc: 
         logging.error("Failed to retrieve alternate profile information from %s.\n%s", url, exc)
-        return {}   # Return blank list to avoid internal server error.
\ No newline at end of file
+        return {}   # Return blank list to avoid internal server error.
+    
+    
+    
+def get_alt_profile_objects(collection: Dict, alt_profiles: Dict) -> Dict:
+    """Generate Profile objects for all alt profiles.
+    
+    Args:
+        collection (Dict): Dict representing collection data.
+        alt_profiles(Dict): Dict of alt profiles { uri : {profile_data}, ...}.
+        
+    Returns:
+        Dict: Dict of Profile objects representing each alternate profile.
+            { profile_name: ProfileObject, ... }
+    """
+    profiles = {}
+    for url, alt in alt_profiles.items():
+        if (
+            url in collection["conforms_to"]["value"]
```

Comment:
For safety I'd check that collection has "conforms_to" before this. Not sure if that's a key that can be missing, but code elsewhere does check for it. Also, the braces here feel unnecessary, could be just fit on one line